### PR TITLE
adding price inspector to search for missing votes info

### PR DESCRIPTION
**Pull request by gsk967**
# Oracle Price Votes Inspector

## Configuration 

1. Add validator's info for checking missing oracle votes
2. Add a window parameter to specify the number of blocks to check from the latest height

### Ex: 
```json
{
    "validators": [
        {
            "validator":"umeevaloper1wusrw8nzgs6d8d47xqej5xqttvsu2geftfgu5f",
            "moniker":"Stakewolle.com | Auto-compound"
        },
        {
            "validator":"umeevaloper16kukyg8l6qse6n2rq24qr6qd3ylgnnzmut8v3u",
            "moniker":"Coinpayu"
        }
    ],
    "window": 20,
    "networks": {
        "canon": {
            "rpc": "http://localhost:26657",
            "api": "http://localhost:1317"
        },
        "mainnet": {
            "rpc": "http://umee.rpc.m.stavr.tech:10457",
            "api": "https://api-umee-ia.cosmosia.notional.ventures"
        }
    }
}
``````

### Without build
```
// For the mainnet
$ go run main.go --network mainnet --config config.json

// For the canon
$ go run main.go --network canon --config config.json
```

## With build the binary 
```
// build the binary
$ go build -o ./build/ . 
```
Run the binary 
```
// For the mainnet
$ ./build/price_inspector --network mainnet --config config.json

// For the canon
$ ./build/price_inspector --network canon --config config.json
```

### CODE REVIEW

File: inspector_go/rpc/req.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/types/module/testutil"
+	"github.com/umee-network/price_inspector/config"
+	otypes "github.com/umee-network/umee/v6/x/oracle/types"
+)
+
+func GetAcceptedDenoms(codec testutil.TestEncodingConfig, api string) []string {
```
Comment (robert-zaremba) [inline inspector_go/rpc/req.go:17]: please use `/client` (in the umee repo). Or update it if necessary.

File: inspector_go/rpc/tx_decode.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	otypes "github.com/umee-network/umee/v6/x/oracle/types"
+)
+
+// TxDecoder decodes a base64 encoded transaction data and returns a slice of ValidatorExgRates and an error if the decoding fails.
+func TxDecoder(txData string, codec testutil.TestEncodingConfig) ([]config.ValidatorExgRates, error) {
+	d, err := utils.DecodeBase64(txData)
+	if err != nil {
+		return nil, err
+	}
+	codec = app.MakeEncodingConfig()
```
Comment (robert-zaremba) [inline inspector_go/rpc/tx_decode.go:18]: codec should be an argument.

File: inspector_go/rpc/tx_decode.go [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+	var c txtypes.Tx
+	err = codec.Codec.Unmarshal(d, &c)
+	if err != nil {
+		return nil, err
+	}
```
Comment (robert-zaremba) [inline inspector_go/rpc/tx_decode.go:23]: is there a ready function in SDK to unpack transaction?

**Review by robert-zaremba — COMMENTED** [review]
we should reuse the client package from umee repo

File: inspector_go/rpc/req.go (hunk 1 above)
Comment (gsk967) [reply inspector_go/rpc/req.go]: Updated

File: inspector_go/rpc/tx_decode.go (hunk 3 above)
Comment (gsk967) [reply inspector_go/rpc/tx_decode.go:17]: I don't think so

File: inspector_go/README.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        "canon": {
+            "rpc": "http://localhost:26657",
+            "grpc": "http://localhost:9090"
+        },
+        "mainnet": {
+            "rpc": "http://umee.rpc.m.stavr.tech:10457",
+            "grpc": "umee-grpc.polkachu.com:13690"
+        }
+    }
+}
```
Comment (robert-zaremba) [inline inspector_go/README.md:33]: ca we use yaml as a config?

File: inspector_go/rpc/req.go [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+	logger := log.New(os.Stderr, "chain-client", log.LstdFlags)
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancel()
+
+	queryClient, err := query.NewClient(logger, grpcEndpoint, 15*time.Second)
+	if err != nil {
+		log.Fatalf("Failed to initalize the query client %s and Error:  %s", grpcEndpoint, err.Error())
+	}
+
+	oQueryClient := otypes.NewQueryClient(queryClient.GrpcConn)
```
Comment (robert-zaremba) [inline inspector_go/rpc/req.go:27]: let's move the client creation (lines 22-27) to a separate function

**Review by robert-zaremba — COMMENTED** [review]
pre-approving. let's use yaml for config file

**Review by robert-zaremba — APPROVED** [review]

File: inspector_go/rpc/req.go (hunk 5 above)
Comment (gsk967) [reply inspector_go/rpc/req.go]: OK

File: inspector_go/README.md (hunk 4 above)
Comment (gsk967) [reply inspector_go/README.md]: Moving from `json` to `yaml` config

**Comment by gsk967** [comment]
> pre-approving. let's use yaml for config file

OK

**Pull request merged**